base/abstractarray.jl

```diff
@@ -1469,6 +1469,10 @@ RangeVecIntList{A<:AbstractVector{Int}} = Union{Tuple{Vararg{Union{AbstractRange
 get(A::AbstractArray, i::Integer, default) = checkbounds(Bool, A, i) ? A[i] : default
 get(A::AbstractArray, I::Tuple{}, default) = checkbounds(Bool, A) ? A[] : default
 get(A::AbstractArray, I::Dims, default) = checkbounds(Bool, A, I...) ? A[I...] : default
+# Issue #40809
+get(f::Callable, A::AbstractArray, i::Integer) = checkbounds(Bool, A, i) ? A[i] : f()
+get(f::Callable, A::AbstractArray, I::Tuple{}) = checkbounds(Bool, A) ? A[] : f()
```

Comment:
`get(f, A, ())` and `get(f, A, (1,2))` are slightly weird, although I see why `get(A, (), 0)` and `get(A, (1,2), 0)` sort-of have to be like that. Are we sure symmetry is better than allowing `get(f::Callable, A::AbstractArray, i...)`?

(Also would delete all comments & links to issue, this isn't some exotic bug people may want to look up. And doesn't even mention arrays.)